tests/aws/services/cloudformation/test_template_engine.py

```diff
@@ -234,24 +233,40 @@ def test_cidr_function(self, deploy_cfn_template):
 
         assert deployed.outputs["Address"] == "10.0.0.0/24"
 
+    @pytest.mark.parametrize(
+        "region",
+        [
+            "us-east-1",
+            "us-east-2",
+            "us-west-1",
+            "us-west-2",
+            "ap-southeast-2",
+            "ap-northeast-1",
+            "eu-central-1",
+            "eu-west-1",
+        ],
+    )
     @markers.aws.validated
-    def test_get_azs_function(self, deploy_cfn_template, snapshot):
+    def test_get_azs_function(self, deploy_cfn_template, region, aws_client_factory, snapshot):
         """
         TODO parametrize this test.
         For that we need to be able to parametrize the client region. The docs show the we should be
         able to put any region in the parameters but it doesn't work. It only accepts the same region from the client config
         if you put anything else it just returns an empty list.
         """
+        snapshot.add_transformer(snapshot.transform.regex(region, "<region>"))
 
         template_path = os.path.join(
             os.path.dirname(__file__), "../../templates/functions_get_azs.yml"
         )
 
+        aws_client = aws_client_factory(region_name=region)
```

Comment:
This logic looks good to me.
We are allowing CFn to use custom client for deployment.